### ISSUE TITLE
ProvidePlugin does not support named exports, only default

### ISSUE THREAD

**Issue by piuccio** [webpack/webpack#3859]
[ProvidePlugin](https://webpack.github.io/docs/list-of-plugins.html#provideplugin) can only be used for default exports, the example in the doc is jQuery.

```js
new webpack.ProvidePlugin({
    $: "jquery"
})
```

However if you're using libraries with named exports, like `preact` you're out of luck

Hello world with preact:

```js
import { h, render } from 'preact';

render(
    <span>Hello, world!</span>,
    document.body
);
```

The variable `h` is never used in the source file but it's needed because babel transpiles jsx into `h()` calls.

Ideally one would only write `import { render } from 'preact';` and then use `ProvidePlugin` to inject the import for `h`.

Maybe something like

```js
new webpack.ProvidePlugin({
    h: ["preact", "h"]   // <- module name becomes an array
})
```

The [preact-boilerplate](https://github.com/developit/preact-boilerplate) imports `h` in every file and then defines an eslint rule to disable the unused error `"no-unused-vars": [0, { "varsIgnorePattern": "^h$" }]`.

[rollup-plugin-inject](https://github.com/rollup/rollup-plugin-inject) allows named exports

**Comment by sokra** [comment]
There is an open PR adding this feature.

**Comment by piuccio** [comment]
@sokra thanks, I didn't notice that. I think however you refer to the merged PR #3597 which makes this issue a duplicate of #2864

I guess it just needs to be released, will probably be fixed in whatever version comes after `2.2.0-rc.3`, maybe `rc.4` or maybe `2.2.1`
